core/src/oracle_state.rs

```diff
@@ -136,13 +135,15 @@ impl OraclePool {
     pub fn new() -> std::result::Result<OraclePool, Error> {
         let config = &ORACLE_CONFIG;
         let local_oracle_address = config.oracle_address.clone();
-        let oracle_pool_nft = RefreshContract::new().pool_nft_token_id();
-        let refresh_nft = PoolContract::new().refresh_nft_token_id();
+        let oracle_pool_nft = config.oracle_pool_nft.clone();
+        let refresh_nft = config.refresh_nft.clone();
         let oracle_pool_participant_token_id = config.oracle_pool_participant_token_id.clone();
         let data_point_source = config.data_point_source()?;
 
         let refresh_box_scan_name = "Refresh Box Scan";
-        let datapoint_contract_address = OracleContract::new().ergo_tree();
+        let datapoint_contract_address = OracleContract::new()
+            .with_pool_nft_token_id(oracle_pool_nft.clone())
```

Comment:
Just wanted to check if this is correct. If we change a constant in the script, does it change the address?

Reply:
Yes. It changes the ErgoTree by swapping the value in constants.

Reply:
Got it, thanks. I think there's a couple of other places where we need to set some constants.